contracts/axelar-auth-verifier/src/test.rs

```diff
@@ -80,7 +255,165 @@ fn test_transfer_operatorship() {
+fn transfer_operatorship_fail_empty_signer_set() {
+    let (env, _, client) = setup_env();
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let empty_signer_set = generate_empty_signer_set(&env);
+
+    let empty_operator_set = empty_signer_set.to_xdr(&env);
+
+    // should throw an error, empty signer set
+    let res = client.try_transfer_operatorship(&empty_operator_set);
+    assert!(res.is_err());
+}
+
+#[test]
+fn transfer_operatorship_fail_zero_weight() {
+    let (env, _, client) = setup_env();
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let mut new_signers = generate_signer_set(&env, randint(1, 10));
+
+    let last_index = new_signers.signer_set.signers.len() as u32 - 1;
+
+    // get last signer and modify its weight to zero
+    if let Some(mut last_signer) = new_signers.signer_set.signers.get(last_index) {
+        last_signer.1 = U256::from_u32(&env, 0);
+        new_signers.signer_set.signers.set(last_index, last_signer);
+    }
+
+    let encoded_new_signer_set = new_signers.signer_set.to_xdr(&env);
+
+    // should throw an error, last signer weight is zero
+    let res = client.try_transfer_operatorship(&encoded_new_signer_set);
+    assert!(res.is_err());
+}
+
+#[test]
+fn transfer_operatorship_fail_zero_threshold() {
+    let (env, _, client) = setup_env();
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let mut new_signers = generate_signer_set(&env, randint(1, 10));
+
+    // set the threshold to zero
+    new_signers.signer_set.threshold = U256::from_u32(&env, 0);
+
+    let encoded_new_signer_set = new_signers.signer_set.to_xdr(&env);
+
+    // should error because the threshold is set to zero
+    let res = client.try_transfer_operatorship(&encoded_new_signer_set);
+    assert!(res.is_err());
+}
+
+#[test]
+fn transfer_operatorship_fail_low_total_weight() {
+    let (env, _, client) = setup_env();
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let mut new_signers = generate_signer_set(&env, randint(1, 10));
+
+    let one = U256::from_u32(&env, 1);
+
+    let total_weight = new_signers
+        .signer_set
+        .signers
+        .iter()
+        .map(|(_, weight)| weight)
+        .reduce(|acc, weight| acc.add(&weight))
+        .expect("Empty signers");
```

Comment:
```suggestion
        .unwrap();
```
since a panic is never expected, unwrap is fine

Reply:
The problem is that during testing it did seem to panic with an empty signer set which I was explaining [above](https://github.com/axelarnetwork/axelar-cgp-soroban/pull/8#discussion_r1518268717). Let me know if I'm missing something here

Reply:
change the test generation so empty signer set can never be created